### DISCUSSION TITLE
Is REST V3 / GraphQL v4 query real-time?

### DISCUSSION THREAD

**Discussion by louisluhao** [community/community#24565] [Apps, API and Webhooks]
<p>I suspect that both are not real-time but somehow cached. <br>
Does anybody know? I can’t find this in the document.  </p>
<p>Let me <em>elaborate a little</em> bit: <br>
I have a service that subscribed to a repo’s webhook event. <br>
When a  <strong>push</strong>  event is coming in, I then query GitHub through REST V3 / GraphQL v4. <br>
To my surprise that the service failed to retrieve info from either REST or GraphQL (Get PR # from commitSHA). <br>
I then added a sleep(1s) in my service, this time it works. </p>
<p>Thus lead to the question above. </p>

**Comment by louisluhao** [comment]
<p>There is a similar case <a href="https://github.community/t/some-webhooks-events-are-published-and-received-before-being-applied-to-requestable-data/14176">https://github.community/t5/GitHub-API-Development-and/Some-webhooks-events-are-published-and-received-before-being/m-p/27397#M2257</a></p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/louisluhao">@louisluhao</a>,</p>
<p>Thanks for writing in about this, I’m sharing the reply from our support team as it might be helpful to another user with a similar concern. In the meantime, we’re looking forward to seeing you around!</p>
<p>“API queries should be real time. However, it sounds like there was a required background job needed to update the <code>merge_commit_sha</code> and you happen to run the query before it was completed. Therefore you were unable to obtain the data, but when you added the <code>sleep</code>, the job was finished and gave you a response.”</p>
